File: matlab/ex4/randInitializeWeights.m
function init_theta = randInitializeWeights(input_layer , output_layer)
%% 函数功能：初始化层与层之间的权重，输入：L_in，输出：L_out
epsilon_init = sqrt(6) / sqrt(input_layer + output_layer);
init_theta = rand( output_layer,input_layer+1) * 2* epsilon_init - epsilon_init;

end

File: python/ex4/randInitializeWeights.py
import numpy as np

# 初始化网络参数
def rand_init_weights(L_in,L_out):
	epsilon = np.sqrt(6) / np.sqrt(L_in + L_out)
	init_theta = np.random.random((L_out,L_in+1)) * 2*epsilon - epsilon

	return init_theta
